### PR TITLE
fix: parse statusText from the extraInfo event

### PR DESCRIPTION
**Pull request by OrKoN**
Issues: #7458

### CODE REVIEW

**Comment by OrKoN** [comment]
@josepharhar PTAL (extracted your change from #7640)

**Review by josepharhar — APPROVED** [review]

**Comment by josepharhar** [comment]
Does this include the test I added with empty status text?

**Comment by OrKoN** [comment]
Overlooked it: I am gonna add it in a moment.

File: src/common/HTTPResponse.ts [hunk 1]
```diff
@@ -94,6 +95,22 @@ export class HTTPResponse {
...
+  /**
+   * @internal
+   */
+  _parseStatusTextFromExtrInfo(
+    extraInfo: Protocol.Network.ResponseReceivedExtraInfoEvent | null
+  ): string | undefined {
+    if (!extraInfo || !extraInfo.headersText) return;
+    const firstLine = extraInfo.headersText.split('\r', 1)[0];
+    if (!firstLine) return;
+    const match = firstLine.match(/[^ ]* [^ ]* (.*)/);
```
Comment (sadym-chromium) [inline src/common/HTTPResponse.ts:107]: It would be great to have unit testы on this parsing.

**Comment by sadym-chromium** [comment]
It would be really great to have the parsing unit-tested. If you have some strong reasons not to do it, please clarify them and go ahead with merging.

**Review by sadym-chromium — APPROVED** [review]

File: src/common/HTTPResponse.ts (hunk 1 above)
Comment (OrKoN) [reply src/common/HTTPResponse.ts:107]: yeah, the reason is that Puppeteer does not have unit tests. The tests that are called "unit" are added with this PR (one existing test and one empty status text test).
